### PR TITLE
validates not allow deficient source

### PR DESCRIPTION
**Pull request by abe-phdev**

### CODE REVIEW

File: app/models/transaction.rb [hunk 1]
```diff
@@ -38,6 +39,22 @@ def actualized?
 
   private
 
+  def source_amount
+    source_code = Entity.find(source_account_id).code
+    @last_data.select { |code, amount| code == source_code.downcase }
+  end
+
+  def amount_sufficiency
```
Comment (rusbal) [inline app/models/transaction.rb]: Change `amount_sufficiency` to `source_sufficiency`

**Review by rusbal — CHANGES_REQUESTED** [review]
Good work but not enough.  The problem is race condition where more than one transaction is being created at the same time.
Let's mob program on this tomorrow.

**Review by rusbal — CHANGES_REQUESTED** [review]
As discussed

```ruby
  # Save as app/models/not_enough_source.rb
  class NotEnoughSource < StandardError
  end

  def do_the_transfer
    transfer_now
  rescue NotEnoughSource => error
    errors.add(:source_account_id, error.message)
  end

  def transfer_now
    ActiveRecord::Base.transaction do
      save_transfer
      source_amount = basahin_muli_sa_db_after_ng_save

      if source_amount.negative?
        raise NotEnoughSource.new "Insufficient funds on source account"
      end
    end
  end
```

**Comment by rusbal** [comment]
As discussed
```ruby
  def create
    ActiveRecord::Base.transaction do

      #-------- BEGIN (All of the below will be undone if not enough source)  --------
      @transaction = Transaction.new(transaction_params)

      if @transaction.transaction_type&.expense_category_id?
        @transaction.expense_account = @transaction.transaction_type.expense_category
        @transaction.fee = @transaction.transaction_type.expense_category.transaction_fee
      end

      if @transaction.save
        #-------- END (All of the above will be undone if not enough source)  --------
        source_amount = tran.summary.values[tran.source_account.code.downcase]
        raise NotEnoughSource.new "Insufficient funds on source account" if source_amount.negative?

        notice = "#{@transaction.actualized? ? '' : 'Draft'} Transaction was successfully created."
        redirect_to root_path, notice: notice
      else
        set_tran_types_for_frontend
        render :edit, status: :unprocessable_entity
      end

    end
  rescue NotEnoughSource => error
    @transaction.errors.add(:source_account_id, error.message)
  end
```

**Review by Don-PhDev — APPROVED** [review]

**Review by rusbal — APPROVED** [review]
🚢 :shipit:

**Pull request merged**